feeds/wlan-ap-consumer/opensync/patches/810-cm2-gateway-manipulate-AW_Bluetooth_Config.patch

```diff
@@ -0,0 +1,116 @@
+Index: opensync-2.0.5.0/src/cm2/src/cm2_event.c
+===================================================================
+--- opensync-2.0.5.0.orig/src/cm2/src/cm2_event.c
++++ opensync-2.0.5.0/src/cm2/src/cm2_event.c
+@@ -537,6 +537,17 @@ start:
+             if (!cm2_is_extender()) {
+                 LOGD("%s Skip onboarding process target_type = %d",
+                      __func__, g_state.target_type);
++
++                /*
++                 * For Gateway only mode.
++                 * Gateway doesn't not advertise BLE onboarding message.
++                 * In Extender's case, AW_Bluetooth_Config is updated along with
++                 * Connection_Manager_Uplink. Since Gateway mode doesn't update
++                 * Connection_Manager_Uplink, AW_Bluetooth_Config is not update
++                 * as well. Therefore, we have to update here.
++                 */
++                cm2_ble_onboarding_apply_config();
++
+                 cm2_set_state(true, CM2_STATE_OVS_INIT);
+             } else {
+                 cm2_extender_init_state();
+@@ -688,6 +699,12 @@ start:
+                 // disconnect
+                 g_state.connected = false;
+                 cm2_ovsdb_set_Manager_target("");
++
++                /* For Gateway only mode. */
++                if (!cm2_is_extender()) {
++                    cm2_ble_onboarding_set_status(false, BLE_ONBOARDING_STATUS_CLOUD_OK);
++                    cm2_ble_onboarding_apply_config();
++                }
+             }
+             if (!g_state.connected)
+             {
+@@ -753,6 +770,12 @@ start:
+                 g_state.connected = false;
+                 cm2_ovsdb_set_Manager_target("");
+                 cm2_write_current_target_addr();
++
++                /* For Gateway only mode. */
++                if (!cm2_is_extender()) {
++                    cm2_ble_onboarding_set_status(false, BLE_ONBOARDING_STATUS_CLOUD_OK);
++                    cm2_ble_onboarding_apply_config();
++                }
+             }
+ 
+             if (g_state.connected)
+@@ -769,6 +792,16 @@ start:
+                 if (cm2_is_extender())
+                     cm2_connection_req_stability_check(LINK_CHECK | ROUTER_CHECK | INTERNET_CHECK);
+ 
++                /* For Gateway only mode. */
++                if (!cm2_is_extender() && cm2_is_connected_to(CM2_DEST_MANAGER)) {
++                    /* if the Gateway is connected to the cloud, enable all status */
++                    cm2_ble_onboarding_set_status(true, BLE_ONBOARDING_STATUS_ETHERNET_LINK);
++                    cm2_ble_onboarding_set_status(true, BLE_ONBOARDING_STATUS_ETHERNET_BACKHAUL);
++                    cm2_ble_onboarding_set_status(true, BLE_ONBOARDING_STATUS_ROUTER_OK);
++                    cm2_ble_onboarding_set_status(true, BLE_ONBOARDING_STATUS_INTERNET_OK);
++                    cm2_ble_onboarding_set_status(true, BLE_ONBOARDING_STATUS_CLOUD_OK);
```

Comment:
Can we put the status bit mask in one set_status function call? It looks to me there is redundancy cm2_ble_onboarding_set_status(), so it will be cm2_ble_onboarding_set_status(true, BLE_ONBOARDING_STATUS_ETHERNET_LINK |
BLE_ONBOARDING_STATUS_ETHERNET_BACKHAUL |
BLE_ONBOARDING_STATUS_ROUTER_OK |
....);

Reply:
It's because `cm2_ble_onboarding_set_status()` set ble_status bit by bit.
Therefore, we have to invoke `cm2_ble_onboarding_set_status()` several times.
One feasible way is to modify g_state.ble_status directly.
```
void cm2_ble_onboarding_set_status(bool state, cm2_ble_onboarding_status_t status)
{
    if (state)
        g_state.ble_status |= (1 << status);
    else
        g_state.ble_status &= ~(1 << status);

    LOGI("Set BT status = %x", g_state.ble_status);
}
```

Reply:
Ok, i see. so it have to set one by one. @@